[hop] make materialize_as_graph disable pre-existing dispatch modes

ydwu4 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #161220

For materializing_as_subgraph, we just want to trace a graph. The handling of different modes should register their own logic.

ydwu4 committed:
[hop] make materialize_as_graph disable pre-existing dispatch modes

[ghstack-poisoned]

ydwu4 committed:
Update on "[hop] make materialize_as_graph disable pre-existing dispatch modes"


For materializing_as_subgraph, we just want to trace a graph. The handling of different modes should register their own logic. 



[ghstack-poisoned]

ydwu4 committed:
Update on "[hop] make materialize_as_graph disable pre-existing dispatch modes"


For materializing_as_subgraph, we just want to trace a graph. The handling of different modes should register their own logic. 



[ghstack-poisoned]

Lucaskabela reviewed (approved):
CI looks good here :) Also seems to be important for using HOPs in decomp, so approving

ydwu4 committed:
Update on "[hop] make materialize_as_graph disable pre-existing dispatch modes"


For materializing_as_subgraph, we just want to trace a graph. The handling of different modes should register their own logic. 



[ghstack-poisoned]
